File: Backend/src/main/java/com/data/backend/service/RoomService.java
package com.data.backend.service;

import com.data.backend.dto.req.RoomSearchRequest;
import com.data.backend.dto.resp.PageResponse;
import com.data.backend.dto.resp.RoomResponse;

public interface RoomService {
    PageResponse<RoomResponse> searchRooms(RoomSearchRequest request);
    RoomResponse getRoomById(Long id);
}

File: Frontend/services/roomService.ts
import { apiClient } from './api';
import type {
  RoomResponse,
  RoomSearchRequest,
  PageResponse,
  ApiResponse,
} from '@/types';

class RoomService {
  private api = apiClient.instance;

  async searchRooms(params: RoomSearchRequest): Promise<PageResponse<RoomResponse>> {
    const response = await this.api.get<ApiResponse<PageResponse<RoomResponse>>>('/rooms/search', {
      params,
    });
    return response.data.data;
  }

  async getRoomById(id: number): Promise<RoomResponse> {
    const response = await this.api.get<ApiResponse<RoomResponse>>(`/rooms/${id}`);
    return response.data.data;
  }
}

export const roomService = new RoomService();


